### PR TITLE
[Spring] Support war file deployment

### PR DESCRIPTION
**Pull request by yilims**
### Related command
az spring app deploy
az spring app deployment create
add new arg --server-version to support war file deployment in standard tier.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Spring

File: src/spring/azext_spring/_deployment_source_factory.py [hunk 1]
```diff
@@ -51,6 +51,39 @@ def fulfilled_options_from_original_source_info(self, deployment_resource,
...
+            version=version
+        )
+
+    def fulfilled_options_from_original_source_info(self, deployment_resource,
+                                                    jvm_options=None, runtime_version=None, **_):
+        if all(x is None for x in [jvm_options, runtime_version]):
+            return {}
+        original_source = deployment_resource.properties.source
+        return {
+            'jvm_options': jvm_options if jvm_options is not None else original_source.jvm_options,
```
Comment (yuwzho) [inline src/spring/azext_spring/_deployment_source_factory.py:80]: miss server_version?

**Review by yuwzho — CHANGES_REQUESTED** [review]

File: src/spring/azext_spring/_validators.py [hunk 2]
```diff
@@ -755,3 +755,8 @@ def validate_managed_environment(namespace):
         managed_environment = parse_resource_id(managed_environment_id)
         if managed_environment['namespace'].lower() != 'microsoft.app' or managed_environment['type'].lower() != 'managedenvironments':
             raise InvalidArgumentValueError('--managed-environment {0} is not a valid Container App Environment resource ID'.format(managed_environment_id))
+
+
+def validate_server_version(cmd, namespace):
+    if is_enterprise_tier(cmd, namespace.resource_group, namespace.service):
```
[the remark this answers is not in the document]
Comment (yilims) [reply src/spring/azext_spring/_validators.py]: updated

File: src/spring/azext_spring/_utils.py [hunk 3]
```diff
@@ -40,8 +40,21 @@ def _get_upload_local_file(runtime_version, artifact_path=None, source_path=None
     return file_type, file_path
 
 
+def _get_file_ext(artifact_path):
+    if artifact_path:
+        return os.path.splitext(artifact_path)[-1]
+    else:
+        return ""
```
[the remark this answers is not in the document]
Comment (yilims) [reply src/spring/azext_spring/_utils.py]: updated

File: src/spring/azext_spring/_deployment_source_factory.py (hunk 1 above)
Comment (yilims) [reply src/spring/azext_spring/_deployment_source_factory.py:80]: updated

**Review by yuwzho — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
Please take a look at this comment https://github.com/Azure/azure-cli-extensions/pull/6723#issuecomment-1704942126

**Comment by yilims** [comment]
> Please take a look at this comment [#6723 (comment)](https://github.com/Azure/azure-cli-extensions/pull/6723#issuecomment-1704942126)

Thanks @zhoxing-ms, I have add description for this PR:

### Related command
az spring app deploy
az spring app deployment create
add new arg --server-version to support war file deployment in standard tier.

**Comment by zhoxing-ms** [comment]
Please write the description of changes which can be perceived by customers into **HISTORY.rst*.
If you want to release a new extension version, please update the version in **setup.py** as well.

**Comment by yilims** [comment]
Thanks @zhoxing-ms, I have updated HISTORY.rst

**Comment by zhoxing-ms** [comment]
> If you want to release a new extension version, please update the version in setup.py as well.

How about the `setup.py` file?

File: src/spring/HISTORY.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 Release History
 ===============
+1.15.0
```
Comment (jiec-msft) [inline src/spring/HISTORY.md:3]: Do we need to also update the version in `setup.py`?

**Comment by yilims** [comment]
> > If you want to release a new extension version, please update the version in setup.py as well.
> 
> How about the `setup.py` file?

@zhoxing-ms, I don't want to release a new version for this PR. There's other new features developed by other folks, will be released with other features together.

File: src/spring/HISTORY.md (hunk 4 above)
Comment (yilims) [reply src/spring/HISTORY.md:3]: No, the PR will be released with other PRs together.

Comment (zhoxing-ms) [inline src/spring/HISTORY.md:3]: Suggest using `upcoming` instead of `1.15.0`, as this version is not currently planned for immediate release

Comment (yilims) [reply src/spring/HISTORY.md]: updated version to upcoming

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**